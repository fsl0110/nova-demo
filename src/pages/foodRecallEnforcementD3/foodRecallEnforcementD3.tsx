import React from "react";
import { Helmet } from "react-helmet";
import { Text } from "../../components";
import { ReportsChartD3Connected as ReportsChartD3 } from "./reportsChartD3/reportsChartD3";
import { InitiatorsChartD3Connected as InitiatorsChartD3 } from "./initiatorsChartD3/initiatorsChartD3";
import text from "../../utils/data/text.json";

export class FoodRecallEnforcementD3 extends React.PureComponent {
  render() {
    return (
      <>
        <Helmet>
          <title>Food recall enforcement reports since 2012"</title>
        </Helmet>
        <div className="page">
          <Text
            topicTag="h1"
            topic={text.reports.topic}
            text={text.reports.text}
          />
          <ReportsChartD3 title={text.reports.chartTitle} />
          <div className="flex-two-column">
            <Text
              className="item"
              topicTag="h2"
              topic={text.initiators.topic}
              text={text.initiators.text}
            />
            <InitiatorsChartD3
              title={text.initiators.chartTitle}
              className="item"
            />
          </div>
        </div>
      </>
    );
  }
}

export default FoodRecallEnforcementD3;
