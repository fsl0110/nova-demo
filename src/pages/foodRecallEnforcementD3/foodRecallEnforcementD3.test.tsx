import React from "react";
import { shallow } from "enzyme";
import toJson from "enzyme-to-json";
import { FoodRecallEnforcementD3 } from "./foodRecallEnforcementD3";

it("renders toMatchSnapshot", () => {
  const mock = shallow(<FoodRecallEnforcementD3 />);
  expect(toJson(mock)).toMatchSnapshot();
});
