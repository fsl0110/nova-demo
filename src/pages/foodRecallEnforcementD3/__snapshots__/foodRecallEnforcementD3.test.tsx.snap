// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders toMatchSnapshot 1`] = `
<div
  className="page"
>
  <Text
    text="This is the openFDA API endpoint for all food product recalls monitored by the FDA. When an FDA-regulated product is either defective or potentially harmful, recalling that product—removing it from the market or correcting the problem—is the most effective means for protecting the public."
    topic="Food recall enforcement reports since 2012"
    topicTag="h1"
  />
  <Connect(ReportsChartD3)
    title="Enforcement reports over time"
  />
  <div
    className="flex-two-column"
  >
    <Text
      className="item"
      text="Manufacturers and/or distributors may initiate a recall at any time to fulfill their responsibility to protect the public health from products that present a risk of injury or gross deception, or are otherwise defective. Firms may also initiate a recall following notification of a problem by FDA or a state agency, in response to a formal request by FDA, or as ordered by FDA."
      topic="The vast majority of food recalls are voluntary"
      topicTag="h2"
    />
    <Connect(InitiatorsChartD3)
      className="item"
      title="Enforcement initiators"
    />
  </div>
</div>
`;
